lib/runner/extensions/waterfall.command.js

```diff
@@ -41,6 +42,28 @@ extractSNR = function (executions, previous) {
     return snr;
 };
 
+/**
+ * Returns the data for the given iteration
+ *
+ * @function getIterationData
+ * @param {Array} data - The data array containing all iterations' data
+ * @param {Number} iteration - The iteration to get data for
+ * @return {Any} - The data for the iteration
+ */
+getIterationData = function (data, iteration) {
+    if (!Array.isArray(data) || data.length < 1) {
```

Comment:
Don't think this check is required at this point.
`data` should be normalized before initializing any command.

Reply:
For the case when run is created not by `runner.run`, but by `new Run` by the user?